src/player.h

```diff
@@ -496,7 +478,7 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         /**
          * Returns an explanation for why the player would miss a melee attack.
          */
-        const char *get_miss_reason();
+        const char **get_miss_reason();
```

Comment:
I know this change comes from the new weighted_list class (which I actually like very much), but there should be no need to change the signature of this function. Change instead its implementation, it should still return only a pointer to char.

That means: instead of changing all the places that *use* `get_miss_reason`, you only change the one line in `get_miss_reason` that returns the pointer.